users/views.py

```diff
@@ -0,0 +1,188 @@
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.authtoken.models import Token
+from rest_framework.decorators import authentication_classes, permission_classes
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from .models import User
+
+
+class UserSerializer(serializers.ModelSerializer):
+    """
+    Serializer for the User model
+    """
+
+    class Meta:
+        """
+        Tells the models about which fields of the model to include in parsed response/request json.
+        """
+
+        model = User
+        fields = ["email", "name", "is_ative", "is_staff", "date_joined"]
+
+
+class LoginUser(APIView):
+    """
+    Logs a User and returns a Token
+    """
+
+    def post(self, request):
+        """
+        This functions validates the credentials and logs a user in
+        Args:
+            request(HttpRequest): Value containing request data
+        Returns:
+            (Response): A json object containing message, code and token
+        """
+        response = {
+            "message": "Wrong email or password",
+            "code": status.HTTP_400_BAD_REQUEST,
+        }
+        try:
+            email = request.POST["email"]
+            password = request.POST["password"]
+            user = get_object_or_404(User, email=email)
+            if check_password(password, user.password):
+                response["message"] = "Logged in successfully"
+                response["code"] = status.HTTP_200_OK
+                if not Token.objects.filter(user=user).exists():
+                    token = Token.objects.create(user=user)
+                else:
+                    token = Token.objects.get(user=user)
+                response["token"] = token.key
+        except KeyError:
+            response["message"] = "Bad Request"
+        except Http404:
+            response["message"] = "User does not exist"
+
+        return Response(response)
+
+
+class UserAPIView(APIView):
+    """
+    User Management API
+    post to add user
+    put to login user
+    patch to update user
+    delete to remove user
+    """
+
+    def post(self, request):
+        """
+        This function adds a user to db
+        Args:
+            request(HttpRequest): Value containing request data
+        Returns:
+            (Response): A json object containing message and code
+        """
+        response = {}
+        try:
+            name = request.POST["name"]
+            email = request.POST["email"].lower()
+            password = request.POST["password"]
+            validate_password(password)
+            validate_email(email)
+            unique_email = not User.objects.filter(email=email).exists()
+            valid_name = name not in ("", " ")
+            if valid_name and unique_email:
+                User.objects.create_user(email=email, password=password, name=name)
```

Comment:
The above validation logic should go to the manager class instead as that would be a common point of reference throughout the application for user-related operations. When some error occurs in validation the error should be returned by the Manager.

Reply:
2. Instead of calling create_user, you can define your `AUTH_USER_MODEL` in Django settings and then use something like 
```
from django.contrib.auth import get_user_model
User = get_user_model()

```